test/test_common/environment.cc

```diff
@@ -121,6 +121,13 @@ std::string TestEnvironment::substitute(const std::string str) {
 std::string TestEnvironment::temporaryFileSubstitute(const std::string& path,
                                                      const PortMap& port_map,
                                                      Network::Address::IpVersion version) {
+  return temporaryFileSubstitute(path, ParamMap(), port_map, version);
+}
+
+std::string TestEnvironment::temporaryFileSubstitute(const std::string& path,
```

Comment:
I don't want to pick on this change, as the problem was building up before, but we've reached the point where my initial hack in `temporaryFileSubstitute` to do some simple regex in a Jinja-style is now not scaling all that well. We can iterate over the same config files probably O(10-100) times now, which doesn't seem great. I don't think you need to fix it, but maybe put in a `TODO(htuch): Switch to a single pass variable substitution.`.